### PR TITLE
Accountt db base functionality

### PR DESCRIPTION
**Pull request by crawter**
Accountt db base functionality

Created User table and all related functionality

package pkg/accountdb/user contains all stuff related to user entity

baseContract, baseDbo and baseRepository will be updated in future

package pkg/accountdb/db will be used to store accountdb.db3 file.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -103,12 +103,14 @@ require (
...
 	github.com/garyburd/redigo v1.0.1-0.20170216214944-0d253a66e6e1 // indirect
 	github.com/go-ole/go-ole v1.2.1 // indirect
+	github.com/google/uuid v1.0.0
 	github.com/hanwen/go-fuse v0.0.0-20181011180456-b760b55765be
 	github.com/inconshreveable/mousetrap v1.0.0 // indirect
 	github.com/mattn/go-colorable v0.0.9 // indirect
 
 	github.com/minio/minio v0.0.0-20180508161510-54cd29b51c38
 	github.com/mitchellh/mapstructure v1.1.1 // indirect
+	github.com/pkg/errors v0.8.0
```
Comment (egonelbre) [inline go.mod]: We shouldn't introduce another errors package.

File: pkg/accountdb/baseContract.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package accountdb
+
+// Base data contract for all entities.
+// Describes all common columns and stored procedures.
+type BaseContract interface {
```
Comment (egonelbre) [inline pkg/accountdb/baseContract.go]: Let's not implement an anemic data model. Let's use dbx for consistency.

File: pkg/accountdb/dbfactory.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+// strings, messages, etc
+var (
+	notInitializedErrorMsg = "DbFactory is not initialized!.\nUse NewDbFactory function to initialize it."
+)
+
+// db instance.
+var db *dbFactory = nil
+
+// NewDbFactory initializes new dbFactory instance and returns pointer on it.
+func NewDbFactory(connstr, driver string) *dbFactory {
```
Comment (egonelbre) [inline pkg/accountdb/dbfactory.go]: The factory doesn't make here sense.

File: pkg/accountdb/dbfactory.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+	driver string
+}
+
+// strings, messages, etc
+var (
+	notInitializedErrorMsg = "DbFactory is not initialized!.\nUse NewDbFactory function to initialize it."
+)
+
+// db instance.
+var db *dbFactory = nil
```
Comment (egonelbre) [inline pkg/accountdb/dbfactory.go]: No global variables.

File: pkg/accountdb/user/contract.go [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+
+	colFirstName: "FirstName",
+	colLastName: "LastName",
+	colEmail: "Email",
+	colPassword: "Password",
+
+	queryGet: `SELECT * FROM %s  
+                   WHERE %s = $1`,
+
+	queryGetByCredentials: `SELECT * FROM %s
```
Comment (egonelbre) [inline pkg/accountdb/user/contract.go]: Do not use fmt.Sprintf, it's an easy way to introduce sql injections.

File: pkg/accountdb/user/repository_test.go [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+	}
+
+	defer func() {
+		conn, _ := db.GetConnection()
+		conn.Close()
+	}()
+
+	repository := NewRepository()
+
+	cases := []struct {
```
Comment (egonelbre) [inline pkg/accountdb/user/repository_test.go]: This cases array doesn't make sense. Use sub-tests instead, if you want separation. Or just write them in blocks.

File: pkg/accountdb/baseContract.go [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+	TableName() string
+
+	// common columns
+	Id() string
+	CreationDate() string
```
Comment (egonelbre) [inline pkg/accountdb/baseContract.go]: CreationDate should be time.Time

File: pkg/accountdb/baseContract.go [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+
+package accountdb
+
+// Base data contract for all entities.
+// Describes all common columns and stored procedures.
+type BaseContract interface {
+	TableName() string
+
+	// common columns
+	Id() string
```
Comment (egonelbre) [inline pkg/accountdb/baseContract.go]: ID should be specific to a thing.

File: pkg/accountdb/user/contract.go (hunk 5 above)
Comment (crawter) [reply pkg/accountdb/user/contract.go]: I use Sprintf only to format column and table names. No values passed here.

File: pkg/accountdb/baseContract.go (hunk 7 above)
Comment (crawter) [reply pkg/accountdb/baseContract.go]: CreationDate returns column name

File: pkg/accountdb/user/contract.go (hunk 5 above)
Comment (egonelbre) [reply pkg/accountdb/user/contract.go]: Sure, I understood that, but still an easy way.

**Pull request closed without merging**